hs_composite_resource/models.py

```diff
@@ -0,0 +1,180 @@
+import os
+
+from mezzanine.pages.page_processors import processor_for
+
+from hs_core.models import BaseResource, ResourceManager, resource_processor
+
+from hs_file_types.models import GenericLogicalFile
+
+
+class CompositeResource(BaseResource):
+    objects = ResourceManager("CompositeResource")
+
+    class Meta:
+        verbose_name = 'Composite Resource'
+        proxy = True
+
+    @property
+    def can_be_public_or_discoverable(self):
+        # resource level metadata check
+        if not super(CompositeResource, self).can_be_public_or_discoverable:
+            return False
+
+        # filetype level metadata check
+        for lf in self.logical_files:
+            if not lf.metadata.has_all_required_elements():
+                return False
+
+        return True
+
+    def set_default_logical_file(self):
+        """sets an instance of GenericLogicalFile to any resource file objects of this instance
+        of the resource that is not already associated with a logical file. """
+
+        for res_file in self.files.all():
+            if not res_file.has_logical_file:
+                logical_file = GenericLogicalFile.create()
+                res_file.logical_file_content_object = logical_file
+                res_file.save()
+
+    @property
+    def supports_logical_file(self):
+        """ if this resource allows associating resource file objects with logical file"""
+        return True
+
+    def get_metadata_xml(self, pretty_print=True):
+        from lxml import etree
+
+        # get resource level core metadata as xml string
+        xml_string = super(CompositeResource, self).get_metadata_xml(pretty_print=False)
+        # add file type metadata xml
+
+        # create an etree xml object
+        RDF_ROOT = etree.fromstring(xml_string)
+
+        # get root 'Description' element that contains all other elements
+        container = RDF_ROOT.find('rdf:Description', namespaces=self.metadata.NAMESPACES)
+
+        for lf in self.logical_files:
+            lf.metadata.add_to_xml_container(container)
+
+        return etree.tostring(RDF_ROOT, pretty_print=pretty_print)
+
+    def supports_folder_creation(self, folder_full_path):
+        """this checks if it is allowed to create a folder at the specified path"""
+
+        path_parts = folder_full_path.split("/")
+        # remove the new folder name from the path
+        path_parts = path_parts[:-1]
+        path_to_check = "/".join(path_parts)
```

Comment:
```path_to_check``` can be obtained directly by using ```rfind()``` to find index of the last ```/```, then get substring up to the last ```/```, which is simpler, faster, and more straightforward than using split followed by join.